libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/V2/Conway/Ledger.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE EmptyCase #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE ImpredicativeTypes #-}
+{-# LANGUAGE DerivingVia #-}
```

Comment:
Do you just copy paste your favorite extensions into each module regardless if they are used or not? :rofl:

Reply:
Because it takes time to figure out which are used and not. The right solution is to put them in the cabal file.

Reply:
> The right solution is to put them in the cabal file.

That's the worst solution, because it would enable all of them in ghci for all the transitive packages in the project that depend on this package

> Because it takes time to figure out which are used and not. 

Being lazy is not an excuse :wink: 

In this module for example I can clearly see which ones are not used: none of them are used :smile:

Reply:
That's a silly argument when talking about a draft PR... It makes a lot of sense to cut corners and copy paste the preamble when working to get things done.

By all means, you may call it lazy. But then, aren't comments about the preamble in a draft PR an example of a lazy review?

Reply:
No worries, I wasn't telling you to change anything. I understand that it is a draft. I just made a comment, cause I found it funny to see an empty module with over 30 extensions in it.

That being said, for a final version of a PR I would appreciate if only extensions that are being used in a module are enabled.

Reply:
You're right of course. I'm just a bit touchy today. Bad day that's all.

Though I would prefer to put things in the cabal file, if that makes sense? (At least the uncontroversial extensions)

Reply:
> That's the worst solution, because it would enable all of them in ghci for all the transitive packages in the project that depend on this package

That's not the case, @UlfNorell just tried it sitting next to me.

Reply:
Here is the GHC ticket for reference: https://gitlab.haskell.org/ghc/ghc/-/issues/10827

`stack` will even produce a nice warnigng about this nowadays:

```
> stack repl

Warning: There are Cabal flags for this project which may prevent GHCi from loading your code properly. In some cases it can also load some projects which would otherwise fail to build.
         
         -XOverloadedLists will be used, but it can cause type ambiguity in code not usually compiled with it. It is specified for: foo:lib. But not for: bar:lib.
         
         To resolve, remove the flag(s) from the Cabal file(s) and instead put them at the top of the Haskell files.
```

When I ran into this I spent half a day scratching my head trying to figure this thing out.

I think it might not be a problem for `cabal` since it refuses to load multiple libraries at once, probably due to the same ghci issue:

```
> cabal repl foo bar
Error: cabal: Cannot open a repl for multiple components at once. The targets
'bar' and 'foo' refer to different components..

The reason for this limitation is that current versions of ghci do not support
loading multiple components as source. Load just one component and when you
make changes to a dependent component then quit and reload.
```

In any case, there are other downsides to enabling extensions globally for the whole package especially for a multipackage repository like that. It is not possible to keep track in your head which packages have which extensions enabled globally.

So, please, do not turn on extensions in the cabal file.